Mathlib/Data/Finsupp/Pointwise.lean

```diff
@@ -92,10 +92,9 @@ instance [NonUnitalCommRing β] : NonUnitalCommRing (α →₀ β) :=
   DFunLike.coe_injective.nonUnitalCommRing _ coe_zero coe_add coe_mul coe_neg coe_sub
     (fun _ _ ↦ rfl) fun _ _ ↦ rfl
 
--- TODO can this be generalized in the direction of `Pi.smul'`
--- (i.e. dependent functions and finsupps)
--- TODO in theory this could be generalised, we only really need `smul_zero` for the definition
-instance pointwiseScalar [Semiring β] : SMul (α → β) (α →₀ β) where
+-- TODO(Paul-Lez): add a `DFinsupp` version of this.
+instance pointwiseScalar {M : Type*} [Zero M] [SMulZeroClass β M] [Semiring β] :
```

Comment:
Perhaps worth a comment that this creates an instance diamond for `SMul (α → β) (α →₀ (α → β))` (for the same reason that `Pi.smul'` does).

Reply:
Actually, looking again the original version did not create a typeclass diamond, but this one does!

Reply:
Ah good point - I've added a comment to that effect.